gobblin-utility/src/test/java/org/apache/gobblin/util/PullFileLoaderTest.java

```diff
@@ -139,6 +142,34 @@ public void testRecursiveJobLoading() throws Exception {
     Assert.assertEquals(pullFile.entrySet().size(), 4);
   }
 
+  /**
+   * Tests to verify job written first to the job catalog is picked up first.
+   * @throws Exception
+   */
+  @Test void testJobLoadingOrder() throws Exception {
+    Properties sysProps = new Properties();
+    FileSystem fs = FileSystem.getLocal(new Configuration());
+    Path localBasePath = new Path(Files.createTempDir().getAbsolutePath(), "PullFileLoaderTestDir");
+    fs.mkdirs(localBasePath);
+
+    for (int i=5; i>0; i--) {
+      String job = localBasePath.toString() + "/job" + i + ".conf";
+      PrintWriter writer = new PrintWriter(job, "UTF-8");
+      writer.println("key=job" + i + "_val");
+      writer.close();
+      Thread.sleep(1000);
+    }
+
+    List<Config> configs =
+        loader.loadPullFilesRecursively(localBasePath, ConfigUtils.propertiesToConfig(sysProps), false);
+
+    int i = 5;
+    for (Config config : configs) {
+      Assert.assertEquals(config.getString("key"), "job" + i + "_val");
+      i--;
+    }
```

Comment:
Please also clean up the files created after the tests are down. Preferably in test tear down.

Reply:
done